eBPF Functions (enable/disable threads) (#15214)
netdata/netdata: collectors/ebpf.plugin/README.md

If you or your distribution compiled the kernel with the last combination, your system cannot load shared libraries
required to run `ebpf.plugin`.

## Function

The eBPF plugin has a [function](https://github.com/netdata/netdata/blob/master/docs/cloud/netdata-functions.md) named
`ebpf_thread` that controls its internal threads and helps to reduce the overhead on host. Using the function you

> can run eBPF plugin with all threads disabled and enable them only when to take a look in specific areas is necessary.

Reviewer: ```suggestion
can run the plugin with all threads disabled and enable them only when you want to take a look in specific areas.
```

### List threads

To list all threads status you can query directly the endpoint function:

`http://localhost:19999/api/v1/function?function=ebpf_thread`

It is also possible to query a specific thread adding keyword `thread` and thread name:

`http://localhost:19999/api/v1/function?function=ebpf_thread%20thread:mount`

### Enable thread

It is possible to enable a specific thread using the keyword `enable`:

`http://localhost:19999/api/v1/function?function=ebpf_thread%20enable:mount`

> this will run thread `mount` during 300 seconds (5 minutes). You can specify a specific period appending the period

Reviewer: ```suggestion
this will run thread `mount` during 300 seconds (5 minutes). You can specify a specific period by appending the period
```

after the thread name:

`http://localhost:19999/api/v1/function?function=ebpf_thread%20enable:mount:600`

in this example thread `mount` will run during 600 seconds (10 minutes).

### Disable thread

> It is also possible to stop any thread running using keywork `disable`, for exaample, to disable `cachestat` you can

Reviewer: ```suggestion
It is also possible to stop any thread running using the keyword `disable`. For example, to disable `cachestat` you can
```

request:

`http://localhost:19999/api/v1/function?function=ebpf_thread%20disable:cachestat`

### Debugging threads

> You can verify the impact of threads on host running the

Reviewer: ```suggestion
You can verify the impact of threads on the host by running the
```

[ebpf_thread_function.sh](https://github.com/netdata/netdata/blob/master/tests/ebpf/ebpf_thread_function.sh)
script on your environment.

> You can check the results of to have threads running on your environment in Netdata monitoring section on your

Reviewer: ```suggestion
You can check the results of having threads running on your environment in the Netdata monitoring section on your
```
